lib/logstash/util/java_version.rb

```diff
@@ -0,0 +1,52 @@
+module LogStash::Util::JavaVersion
+  # Print a warning if we're on a bad version of java
+  def self.warn_on_bad_java_version
+    if self.bad_java_version?(self.version)
+      STDERR.puts("!!! Please upgrade your java version, the current version '#{self.version}' may cause problems. We recommend a minimum version of 1.7.0_51")
+    end
+  end
+
+  # Check to see if this is a recommended java version, print a warning to stdout if this is a bad version
+  # Returns the current java version
+  def self.version
+    return nil unless LogStash::Environment.jruby?
+
+    require 'java'
+    java_import "java.lang.System"
+
+    System.getProperty("java.runtime.version")
+  end
+
+  def self.parse_java_version(version_string)
+    return nil if version_string.nil?
+
+    # Crazy java versioning rules @ http://www.oracle.com/technetwork/java/javase/versioning-naming-139433.html
+    # The regex below parses this all correctly http://rubular.com/r/sInQc3Nc7f
+
+    match = version_string.match(/\A(\d+)\.(\d+)\.(\d+)(_(\d+))?(-(.+))?\Z/)
+    major, minor, patch, ufull, update, bfull, build = match.captures
+
+    return {
```

Comment:
I read in the style guide that explicit returns were preferred. I'm 100% with you on implicit return personally. Is the style guide out of date?